### PR TITLE
Migration to vue 3

### PR DESCRIPTION
**Pull request by DorraJaouad**
This PR not to be merged into master but for easier changes reviewing.

Steps done: 

- Update deps 
- Removed breaking changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,12 +39,13 @@
     "dist"
   ],
   "peerDependencies": {
-    "vue": "^2.7.16"
+    "vue": "3"
```
Comment (ShGKme) [inline package.json:42]: `"@vue/compiler-sfc": "^3.4.23"`, so here vue should be also `^3.4.23`, not `3`

File: package.json [hunk 2]
```diff
@@ -39,12 +39,13 @@
...
   },
   "dependencies": {
     "@nextcloud/axios": "^2.4.0",
     "@nextcloud/router": "^3.0.0",
-    "@nextcloud/vue": "^8.11.2",
+    "@nextcloud/vue": "^9.0.0-alpha.2",
+    "@vue/compiler-sfc": "^3.4.23",
```
Comment (ShGKme) [inline package.json:48]: `@vue/compiler-sfc` is not needed as a dependency, it is a sub dependency of `vue` in Vue 3.2+

File: src/collectionstore.js [hunk 3]
```diff
@@ -29,18 +29,18 @@ const state = reactive({
...
 	addCollections(collections) {
-		set(state, 'collections', collections)
+		state.collections = collections
 	},
 	addCollection(collection) {
 		state.collections.push(collection)
 	},
 	removeCollection(collectionId) {
-		set(state, 'collections', state.collections.filter(item => item.id !== collectionId))
+		state.collections= state.collections.filter(item => item.id !== collectionId)
```
Comment (ShGKme) [inline src/collectionstore.js:38]: ESLint:
> Operator '=' must be spaced

File: src/collectionstore.js [hunk 4]
```diff
@@ -29,18 +29,18 @@ const state = reactive({
...
 	},
 	updateCollection(collection) {
 		const index = state.collections.findIndex((_item) => _item.id === collection.id)
 		if (index !== -1) {
-			set(state.collections, index, collection)
+			state.collections.index = collection
```
Comment (ShGKme) [inline src/collectionstore.js:43]:
```suggestion
			state.collections[index] = collection
```

File: package.json (hunk 1 above)
Comment (ShGKme) [reply package.json:42]: Or lower, because it will also work fine on a lower version

Comment (DorraJaouad) [reply package.json:42]: `3.4.23` would be fine

File: package.json [hunk 5]
```diff
@@ -39,12 +39,12 @@
     "dist"
   ],
   "peerDependencies": {
-    "vue": "^2.7.16"
+    "vue": "3.4.23"
```
Comment (Antreesy) [inline package.json:42]:
```suggestion
    "vue": "^3.4.23"
```

**Review by Antreesy — APPROVED** [review]
Rest seems OK, let's wait for Grigorii to re-review it)

File: package.json (hunk 5 above)
Comment (ShGKme) [reply package.json:42]: Agree, it should not be the exact version

**Pull request merged**